pkg/infra/client.go

```diff
@@ -46,43 +47,43 @@ func CreateGRPCClient(node Node) (*comm.GRPCClient, error) {
 }
 
 func CreateEndorserClient(node Node) (peer.EndorserClient, error) {
-	gRPCClient, err := CreateGRPCClient(node)
+	conn, err := DailConnection(node)
 	if err != nil {
 		return nil, err
 	}
-
-	conn, err := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
-	if err != nil {
-		return nil, errors.Wrapf(err, "error connecting to %s", node.Addr)
-	}
-
 	return peer.NewEndorserClient(conn), nil
 }
 
 func CreateBroadcastClient(node Node) (orderer.AtomicBroadcast_BroadcastClient, error) {
-	gRPCClient, err := CreateGRPCClient(node)
+	conn, err := DailConnection(node)
 	if err != nil {
 		return nil, err
 	}
-
-	conn, err := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
-	if err != nil {
-		return nil, errors.Wrapf(err, "error connecting to %s", node.Addr)
-	}
-
 	return orderer.NewAtomicBroadcastClient(conn).Broadcast(context.Background())
 }
 
 func CreateDeliverFilteredClient(node Node) (peer.Deliver_DeliverFilteredClient, error) {
-	gRPCClient, err := CreateGRPCClient(node)
+	conn, err := DailConnection(node)
 	if err != nil {
 		return nil, err
 	}
+	return peer.NewDeliverClient(conn).DeliverFiltered(context.Background())
+}
 
-	conn, err := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
+func DailConnection(node Node) (*grpc.ClientConn, error) {
+	gRPCClient, err := CreateGRPCClient(node)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error connecting to %s", node.Addr)
+		return nil, err
 	}
-
-	return peer.NewDeliverClient(conn).DeliverFiltered(context.Background())
+	var connError error
+	for i := 0; i < 3; i++ {
+		conn, connError := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
```

Comment:
this still create a new `connError` within the for scope

Reply:
fixed for this one.